httpx/utils.py

```diff
@@ -84,9 +84,25 @@ def guess_json_utf(data: bytes) -> typing.Optional[str]:
 
 
 def get_netrc_login(host: str) -> typing.Optional[typing.Tuple[str, str, str]]:
-    try:
-        netrc_info = netrc.netrc(os.environ.get("NETRC"))  # type: ignore
-    except FileNotFoundError:
+    if "NETRC" in os.environ:
+        NETRC_FILES = (os.environ["NETRC"],)
+    else:
+        try:
+            NETRC_FILES = (
+                os.path.expanduser("~/.netrc"),
+                os.path.expanduser("~/_netrc"),
+            )
+        except KeyError:  # pragma: nocover
```

Comment:
Why would this return a `KeyError`?

Reply:
The `requests` codebase has the same case, claiming "os.path.expanduser" can raise a KeyError if no user directory is set. However thinking about it if that ever *was* true then it's likely it would have been treated as a valid bug in Python, so it's unlikely to be true now. The docs state "If the expansion fails or if the path does not begin with a tilde, the path is returned unchanged." so we probably don't want this catch-case.